packages/cli/src/commands/account/list.ts

```diff
@@ -8,29 +8,38 @@ export default class AccountList extends BaseCommand {
 
   static flags = {
     ...BaseCommand.flags,
+    local: flags.boolean({
+      allowNo: true,
+      description:
+        'If set, only show local and hardware wallet accounts. Use no-local to only show keystore addresses.',
+    }),
   }
 
   requireSynced = false
 
   async run() {
-    this.parse(AccountList)
+    const res = this.parse(AccountList)
 
+    // Retreive accounts from the connected Celo node.
+    const nodeAddresses = !res.flags.local ? await this.kit.web3.eth.getAccounts() : []
+
+    // Get addresses from the local wallet.
     const celoProvider: CeloProvider = this.kit.web3.currentProvider as any
-    const addresses = await this.kit.web3.eth.getAccounts()
-    const localAddresses = celoProvider.wallet
-      .getAccounts()
-      .map((value) => toChecksumAddress(value))
-    let localName = 'Local'
-    const res: ParserOutput<any, any> = this.parse()
+    const localAddresses =
+      res.flags.local ?? true
```

Comment:
this is like having the flag `local` with a default: true
If I'm understanding what you wanted to do, basically this gives you the possibility of later, hide the keystore address
I'm not really sure why that's needed. I totally agree on the "only local" because you won't need a node to check your ledger for example, but in the other case IMO it seems the same (remember that the `getAccounts` already do the work of getting the local addresses)